src/main/java/htsjdk/samtools/SAMReadGroupRecord.java

```diff
@@ -90,6 +93,36 @@ public SAMReadGroupRecord(final String id, final SAMReadGroupRecord srcProgramRe
     public String getPlatform() { return getAttribute(PLATFORM_TAG); }
     public void setPlatform(final String platform) { setAttribute(PLATFORM_TAG, platform); }
 
+    /**
+     * @return the List of barcodes associated with this read group or null
+     */
+    public List<String> getBarcodes() {
```

Comment:
I would add a method using a different barcode separator too (in case it is needed by other implementations), and the one without params should use the default `BARCODE_SEPARATOR`

Reply:
I found some FASTQ files with `+` as barcode separator, that when are converted with custom scripts to SAM/BAM the `@RG` tag (per-read) keeps that symbol; thus, some users of ReadTools have asked me to support other separators, which I did with custom code, but as a recommendation in the specs it looks like having a method supporting other implementations will be nice for the library.

Anyway, I am planning to come back in the real near future to htsjk3, where we can take the decisions about supporting the "recommended" specs alone or all of them.